tamr_client/attribute/_attribute.py

```diff
@@ -157,9 +168,15 @@ def create(
     if name in _RESERVED_NAMES:
         raise ReservedName(name)
 
+    if isinstance(parent, UnifiedDataset):
+        raise ValueError(
```

Comment:
For the exception name, I think explicitness is better than concision. So something like `raise CannotCreateAttributesOnUnifiedDataset` is fine by me.

Reply:
@pcattori What do you think, is this a reasonable level of protection to build in?

Currently, `<project_id>` is always an integer, but in the broadest possible generality, the suggested solution could break on "/"-containing project IDs (if that's even possible).

Reply:
Is there a way to refer to attributes via a canonical URL that is not relative to a project nor a UD?

If no: we can do some rudimentary checking of the URL as you suggest. Probably worth doing.
If yes: we'd have do some gymnastics to figure out if its a UA or a project attribute. Probably not worth doing, too complex.

Reply:
There are internal API endpoints that allow specifying a unified attribute by an ID alone, e.g. `343`, but there is no way of identifying an attribute through a versioned URL except for
1. `/v1/datasets/{datasetId}/attributes/{attributeId}`
2. `/v1/projects/{project}/unifiedDataset/attributes/{attributeId}` (an alias of the above)
3. `/v1/projects/{project}/attributes/{attributeId}`
where 2 and 3 are only possible for project/unified dataset attributes.

Using tamr_client as currently designed, it is possible to get a unified attribute `Attribute`, with a standard dataset URL that we can't detect with the simple check, if the user gets the unified dataset by resource ID or name as a regular `Dataset` and then gets the attribute from there.  The simple check only works if the user gets the unified dataset using the `tamr_client.dataset.unified.from_project` function, vs the `tamr_client.dataset.by_resource_id` or `.by_name` functions.

Reply:
Let's skip on doing guardrails here. Let's do the simple thing, get it out there, and then act on any bug reports with more context.